internal/events/cmdcall/cmdcall.go

```diff
@@ -0,0 +1,94 @@
+package cmdcall
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/scriptrun"
+	"github.com/ActiveState/cli/internal/subshell"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+type primeable interface {
+	primer.Outputer
+	primer.Projecter
+	primer.Subsheller
+}
+
+// CmdCall manages dependencies for the handling of events triggered by command
+// calls.
+type CmdCall struct {
+	out      output.Outputer
+	proj     *project.Project
+	subshell subshell.SubShell
+	cmdList  string
+	p        primeable
+}
+
+// New returns a prepared pointer to an instance of CmdCall.
+func New(p primeable, cmdList string) *CmdCall {
+	return &CmdCall{
+		out:      p.Output(),
+		proj:     p.Project(),
+		subshell: p.Subshell(),
+		cmdList:  cmdList,
+		p:        p,
+	}
+}
+
+// Run executes the event handling logic by running any relevant scripts.
+func (cc *CmdCall) Run(t project.EventType) error {
+	logging.Debug("cmdcall")
+
+	if cc.proj == nil {
+		return nil
+	}
+
+	var events []*project.Event
+	for _, event := range cc.proj.Events() {
+		if event.Name() != string(t) {
+			continue
+		}
+
+		scopes, err := event.Scope()
+		if err != nil {
+			return err // TODO: this
+		}
+
+		for _, scope := range scopes {
+			if scope == cc.cmdList {
+				events = append(events, event)
+			}
+		}
+
+	}
+
+	if len(events) == 0 {
+		return nil
+	}
+
+	for _, event := range events {
+		val, err := event.Value()
+		if err != nil {
+			return err // TODO: this
+		}
+
+		ss := strings.Split(val, " ")
+		if len(ss) == 0 {
+			return errors.New("no script defined") // TODO: this
+		}
+
+		if !scriptrun.ProjectHasScript(cc.proj, ss[0]) {
```

Comment:
I feel like `RunScript` should be doing this?

Reply:
Yes, it also does that check. I did this for now to see if we want to skip when the script isn't present, or error out. For instance, if there are 3 events with different scripts, do we want to halt on script 2 if it doesn't exist, or should 1 and 3 be run with 2 being skipped?

Reply:
We should halt as soon as there is an error. I don't see what this check has to do with that though.

Reply:
In the run runner, if the script is not able to be found, it never prints the title. This is a similar check that, minimally, ensures the script exists before running the script itself. Since we want to error out, the check is useless now and will be removed.